Orleans/src/Microsoft.Orleans.Host.Abstractions/StorageProvidersFeature.cs

```diff
@@ -0,0 +1,51 @@
+﻿using Microsoft.Extensions.DependencyInjection;
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace Microsoft.Orleans.Host.Abstractions
+{
+    // this would go into the (optional) storage providers DLL.
+    public interface IStorageProvidersBuilder
+    {
+        ISiloBuilder Silo { get; }
```

Comment:
this way storage providers are an opt-in framework feature that do not require to be bundled in the virtual actors core. You decide whether you want to use storage providers explicitly, and since it's entirely opt-in, you might decide to use an alternative storage providers model entirely, because the framework would be flexible enough to support it.